### PR TITLE
Rid Colossus bidder of OpenrtbBidder.

### PR DESCRIPTION
**Pull request by CTMBNara**

### CODE REVIEW

File: src/main/java/org/prebid/server/bidder/colossus/ColossusBidder.java [hunk 1]
```diff
@@ -1,20 +1,143 @@
...
+        for (Imp imp : imps) {
+            if (Objects.equals(impId, imp.getId())) {
+                if (imp.getBanner() == null && imp.getVideo() != null) {
+                    return BidType.video;
+                }
+                return BidType.banner;
+            }
+        }
+
+        throw new PreBidException(String.format("Failed to find impression for ID: %s", impId));
```
Comment (And1sS) [inline src/main/java/org/prebid/server/bidder/colossus/ColossusBidder.java:141]: Seems like this logic does not exactly match logic present in OpenrtbBidder

File: src/main/java/org/prebid/server/bidder/colossus/ColossusBidder.java [hunk 2]
```diff
@@ -1,20 +1,143 @@
...
+        try {
+            bidResponse = mapper.decodeValue(httpCall.getResponse().getBody(), BidResponse.class);
+        } catch (DecodeException e) {
+            return Result.withError(BidderError.badServerResponse(e.getMessage()));
+        }
+
+        final List<BidderError> errors = new ArrayList<>();
+        final List<BidderBid> bids = extractBids(httpCall.getRequest().getPayload(), bidResponse, errors);
+
+        return Result.of(bids, errors);
```
Comment (And1sS) [inline src/main/java/org/prebid/server/bidder/colossus/ColossusBidder.java:100]: OpenrtbBidder does not collect errors, it just return result of first occured error.

File: src/main/java/org/prebid/server/bidder/colossus/ColossusBidder.java [hunk 3]
```diff
@@ -1,20 +1,143 @@
...
+        this.mapper = Objects.requireNonNull(mapper);
     }
 
     @Override
-    protected Imp modifyImp(Imp imp, ExtImpColossus impExt) {
-        return imp.toBuilder()
-                .tagid(impExt.getTagId())
+    public Result<List<HttpRequest<BidRequest>>> makeHttpRequests(BidRequest bidRequest) {
+        final List<HttpRequest<BidRequest>> requests = new ArrayList<>();
+        final List<BidderError> errors = new ArrayList<>();
```
Comment (And1sS) [inline src/main/java/org/prebid/server/bidder/colossus/ColossusBidder.java:49]: Current implementation of OpenrtbBidder does not store errors.

**Review by And1sS — CHANGES_REQUESTED** [review]

File: src/main/java/org/prebid/server/bidder/colossus/ColossusBidder.java (hunk 3 above)
Comment (CTMBNara) [reply src/main/java/org/prebid/server/bidder/colossus/ColossusBidder.java:49]: It does.

File: src/main/java/org/prebid/server/bidder/colossus/ColossusBidder.java (hunk 2 above)
Comment (CTMBNara) [reply src/main/java/org/prebid/server/bidder/colossus/ColossusBidder.java:100]: Pls, consider go code

File: src/main/java/org/prebid/server/bidder/colossus/ColossusBidder.java (hunk 1 above)
Comment (CTMBNara) [reply src/main/java/org/prebid/server/bidder/colossus/ColossusBidder.java:141]: Pls, consider go code

**Review by And1sS — APPROVED** [review]

**Pull request merged**